src/mono/System.Private.CoreLib/src/System/Array.Mono.cs

```diff
@@ -376,37 +377,38 @@ public static unsafe Array CreateInstance(Type elementType, int[] lengths, int[]
         [MethodImpl(MethodImplOptions.InternalCall)]
         private static extern unsafe void InternalCreate([NotNull] ref Array? result, IntPtr elementType, int rank, int* lengths, int* lowerBounds);
 
-        public object GetValue(int index)
+        private unsafe nint GetFlattenedIndex(ReadOnlySpan<int> indices)
         {
-            if (Rank != 1)
-                ThrowHelper.ThrowArgumentException(ExceptionResource.Arg_Need1DArray);
+            // Checked by the caller
+            Debug.Assert(indices.Length == Rank);
 
-            int lb = GetLowerBound(0);
-            if (index < lb || index > GetUpperBound(0))
-                throw new IndexOutOfRangeException(SR.Argument_IndexOutOfArrayBounds);
-
-            if (GetType().GetElementType()!.IsPointer)
-                throw new NotSupportedException(SR.NotSupported_Type);
-
-            return GetValueImpl(index - lb);
+            nint flattenedIndex = 0;
+            for (int i = 0; i < indices.Length; i++)
+            {
+                int index = indices[i] - GetLowerBound(i);
+                int length = GetLength(i);
+                if ((uint)index >= (uint)length)
```

Comment:
This does exclude the `index < MONO_HANDLE_GETVAL (arr, bounds [i].lower_bound))` check from the old code, which I guess here would just be `index < 0`. Are you confident that won't be hit?

Reply:
The cast to `(uint)` takes care of checking for the `index< 0` condition. We use this trick all over the place in CoreLib.

Reply:
Ahhh, makes sense.